src/transformers/trainer.py

```diff
@@ -368,14 +376,8 @@ def train(self, model_path: Optional[str] = None):
                             if self.tb_writer:
                                 for k, v in logs.items():
                                     self.tb_writer.add_scalar(k, v, global_step)
-                            if _has_wandb:
-                                wandb.log(
-                                    {
-                                        **logs,
-                                        "epoch": global_step
-                                        / (len(train_dataloader) // self.args.gradient_accumulation_steps),
-                                    }
-                                )
+                            if is_wandb_available():
+                                wandb.log(logs, step=global_step)
```

Comment:
did a simpler version of this

Reply:
The only purpose here was to log `epoch` (as a float). When you press the x-axis, you can use "epoch" instead of just the "global_step".
I find it especially useful when I want to compare 2 runs with different batch size for example (let's say if a model is much bigger than another and I cannot fit as many samples in my GPU).

Reply:
Makes sense. Let's look for a way to add it back in a subsequent PR (would be useful for TensorBoard too)

Reply:
uncommented in fd2174664c8879c747ada3e6e0a2486858808421